algorithms/CPlusPlus/Dynamic-Programming/matrix-chain-multiplication.cpp

```diff
@@ -0,0 +1,56 @@
+// Matrix Chain Multiplicatin - Memorization-Approuch
+
+#include <bits/stdc++.h>
+using namespace std;
+
+// Function for matrix chain multiplication
+int matrixChainMulti(int arr[], int i, int j, vector<vector<int>> dp)
```

Comment:
Is it really mandatory to copy vector `dp` every-time you pass it to the function ?

also I see you not are changing the value of `i` and `j` in the function, then wouldn't it be a good idea to make them const ?

```suggestion
int matrixChainMulti(int arr[],const int i,const int j, vector<vector<int>> &dp)
```